azext_iot/central/services/device.py

```diff
@@ -84,6 +85,47 @@ def list_devices(
     return devices
 
 
+def get_device_registration_summary(
+    cmd, app_id: str, token: str, central_dns_suffix="azureiotcentral.com",
+):
+    """
+    Get device registration summary for a given app
+
+    Args:
+        cmd: command passed into az
+        app_id: name of app (used for forming request URL)
+        token: (OPTIONAL) authorization token to fetch device details from IoTC.
+            MUST INCLUDE type (e.g. 'SharedAccessToken ...', 'Bearer ...')
+        central_dns_suffix: {centralDnsSuffixInPath} as found in docs
+
+    Returns:
+        registration summary
+    """
+
+    registration_summary = {
+        DeviceStatus.provisioned.value: 0,
+        DeviceStatus.registered.value: 0,
+        DeviceStatus.unassociated.value: 0,
+        DeviceStatus.blocked.value: 0,
+    }
+
+    url = "https://{}.{}/{}".format(app_id, central_dns_suffix, BASE_PATH)
+    headers = _utility.get_headers(token, cmd)
+
+    while url is not None:
+        response = requests.get(url, headers=headers)
+        result = _utility.try_extract_result(response)
+
+        if "value" not in result:
+            raise CLIError("Value is not present in body: {}".format(result))
```

Comment:
So I know this is from my earlier implementation, but do we want to throw away the whole result if there's an issue?

Or do we want to surface a warning and return the data we did manage to process?

I'd also raise this concern for using the `_utility.try_extract_result(response)` here

Reply:
We can also file a followup item on this

---
In reply to: [428981926](https://github.com/Azure/azure-iot-cli-extension/pull/192#discussion_r428981926) [](ancestors = 428981926)

Reply:
#wontfix  - partial data will not get consistent result and is not useful.